pkg/build/util/generate.go

```diff
@@ -30,3 +32,43 @@ func GenerateBuildFromBuild(build *api.Build) *api.Build {
 		},
 	}
 }
+
+// SubstituteImageReferences replaces references to an image with a new value
+func SubstituteImageReferences(build *api.Build, oldImage string, newImage string) {
+	switch {
+	case build.Parameters.Strategy.Type == api.DockerBuildStrategyType &&
+		build.Parameters.Strategy.DockerStrategy != nil &&
```

Comment:
A build which has StrategyType == api.DockerBuild but nil DockerStrategy is invalid, and the result of invoking SubstituteImageReferences is undefined.  I'd prefer to just hard fail on this - we shouldn't be running SubstituteImageReferences on Invalid objects.